bun/src/binding.ts

```diff
@@ -116,10 +113,15 @@ const lib = dlopen(binaryPath, {
 		],
 		returns: "u32",
 	},
-});
+};
 
-export const binding = lib.symbols;
+// Load the compiled Zig shared library
+const lib = await openLibrary(path.resolve("../bun"), fns);
+export const binding: ConvertFns<typeof fns> = lib.symbols;
 
-export function closeBinding(): void {
+/**
+ * Call this api to close the binding.
+ */
+export async function closeBinding(): Promise<void> {
```

Comment:
this can be a synchronous function (you can just export a const like `export const close = lib.close`)